### PR TITLE
Pv upgrade rails

### PR DESCRIPTION
**Pull request by patrickvinograd**
CVEs against 4.2.5.2. Cascading change to schema_plus_index changes schema.rb formatting.

### CODE REVIEW

**Comment by patrickvinograd** [comment]
Fails in bundle-audit due to ruby-saml, which will be removed in subsequent PR #378 which I want to rebase after this is merged.

**Comment by saneshark** [comment]
@patrickvinograd I noticed that SimpleCov.minimum_coverage is 65, if this gets bumped up to 90, I'm pretty sure test coverage will not be met as a criteria for build passing, but is it worth maybe investigating this further?

I don't have enough back-story to feel 100% comfortable reviewing these changes with test coverage possibly being less than 90%. Happy to review though if its critical we get this through right away. Ping me on slack and lets discuss. Right now it looks like build isn't passing.

File: db/schema.rb [hunk 1]
```diff
@@ -22,27 +22,27 @@
...
+    t.integer  "veteran_id",   :index=>{:name=>"index_affiliations_on_veteran_id"}
   end
 
   create_table "awards", force: :cascade do |t|
     t.string   "title"
     t.string   "organization"
     t.datetime "created_at"
     t.datetime "updated_at"
-    t.integer  "veteran_id",   index: {name: "index_awards_on_veteran_id"}
+    t.integer  "veteran_id",   :index=>{:name=>"index_awards_on_veteran_id"}
```
Comment (saneshark) [inline db/schema.rb]: are these schema changes necessary? They look mostly syntactical. Were these changes the result of rubocop rule preferring hash-rocket?

File: app/views/static_pages/employers.html.erb [hunk 2]
```diff
@@ -5,7 +5,7 @@
 <% unless user_signed_in? && current_user.is_employer? %>
   <div class="header-bar">
     <div class="row">
-      <span class="explanation">Log in to create and manage your account with</span><%= link_to "LinkedIn", user_omniauth_authorize_path(:linkedin), class: "employer-login" %> <%= link_to "Google", user_omniauth_authorize_path(:google_oauth2), class: "employer-login" %></span>
+      <span class="explanation">Log in to create and manage your account with</span><%= link_to "LinkedIn", user_linkedin_omniauth_authorize_path(), class: "employer-login" %> <%= link_to "Google", user_google_oauth2_omniauth_authorize_path(), class: "employer-login" %></span>
```
Comment (saneshark) [inline app/views/static_pages/employers.html.erb]: I don't think you need to specify empty parentheses if there are no arguments.

File: app/views/static_pages/for_job_seekers.html.erb [hunk 3]
```diff
@@ -32,7 +32,7 @@
       </div>
       <div class="small-12 medium-4 columns feature">
         <p>Already have a public profile? Sign in to update it.</p>
-		<%= link_to "Sign in", user_omniauth_authorize_path(:saml), class: "usa-button sign-in", id: "update_profile" %>
+		<%= link_to "Sign in", user_saml_omniauth_authorize_path(), class: "usa-button sign-in", id: "update_profile" %>
```
Comment (saneshark) [inline app/views/static_pages/for_job_seekers.html.erb]: I don't think you need to specify empty parentheses if there are no arguments.

File: app/views/veterans/_form.html.erb [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <script type="text/javascript"  src="https://maps.googleapis.com/maps/api/js?libraries=places"%></script>
 
 <% unless session[:linkedin_profile] %>
-      <%= link_to "<i class='icon-linkedin' aria-hidden='true'></i><span>Auto-fill from LinkedIn</span>".html_safe, user_omniauth_authorize_path(:linkedin_resume), class: "usa-button", id: "linkedin_autofill"%>
+      <%= link_to "<i class='icon-linkedin' aria-hidden='true'></i><span>Auto-fill from LinkedIn</span>".html_safe, user_linkedin_resume_omniauth_authorize_path(), class: "usa-button", id: "linkedin_autofill"%>
```
Comment (saneshark) [inline app/views/veterans/_form.html.erb]: I don't think you need to specify empty parentheses if there are no arguments.

File: db/schema.rb (hunk 1 above)
Comment (patrickvinograd) [inline db/schema.rb]: They were caused by the upgrade of the schema_plus_index Gem. We were locked on a forked version created by Mike Auclair, but upgrading rails caused that not to work and the bugfix in Mike's fork got incorporated into the official Gem.

Comment (patrickvinograd) [inline db/schema.rb]: I think they're all syntactical but I will give it another look-through.

**Comment by ayaleloehr** [comment]
Just chiming in here to add to the record that Patrick and I had talked about this in Slack. Code coverage on this application does not meet our current 90% threshold, but some of this application code will be removed in subsequent PRs and the rest will eventually be ported to vets-api and vets-website, so upping the coverage in this repo is not a priority at this time. 

Glad to have these updates in though!

**Pull request merged**